salt/states/dockerng.py

```diff
@@ -195,7 +195,8 @@ def _compare(actual, create_kwargs, runtime_kwargs):
                           .format(item, actual_ports))
                 log.trace('dockerng.running ({0}): munged desired value: {1}'
                           .format(item, desired_ports))
-                if actual_ports != desired_ports:
+                difference = set(desired_ports).difference(set(actual_ports))
+                if difference:
                     ret.update({item: {'old': actual_ports,
```

Comment:
Are you on freenode? You can find me there, same username.

I've found another problem with using ``set.difference()`` here. This only works if the first set is a superset of the second. For example:

```python
>>> set([1, 2]).difference(set([1]))
set([2])
>>> set([1, 2]).difference(set([1, 2, 3]))
set([])
```